### ISSUE TITLE
outside overshoots

### ISSUE THREAD

**Issue by tbfleming** [andrewhodel/millcrum#2]
``` javascript
var tool = {units:'mm',diameter:10,passDepth:4,step:1,rapid:2000,plunge:100,cut:600,zClearance:5,returnHome:true};
var mc = new Millcrum(tool);
mc.surface(300,300);-517,-11
-502,-6
var polygon = {type:'polygon',points:[
    [0,0],
    [100,0],
    [150,50],
    [100,50],
    [100,70],
    [150,70],
    [150,100],
    [0,100],
]};

mc.cut('outside', polygon, 6, [10,10]);
mc.get();
```

![image](https://cloud.githubusercontent.com/assets/418464/8297861/2a56a19a-192f-11e5-9e04-2a12c75c2a6f.png)

![image](https://cloud.githubusercontent.com/assets/418464/8297906/d761411a-192f-11e5-897d-5580f686bbf9.png)

The usual fix for this is to insert an arc; that prevents the cutter from leaving a gap.

**Comment by andrewhodel** [comment]
If I'm understanding this right, the problem is not that the part is being cut with incorrect dimensions... it's that there's a tool sized chunk taken out of the uncut material that doesn't necessarily need to be?

**Comment by tbfleming** [comment]
Yes. That can create problems when parts are tightly packed to save space.

**Comment by andrewhodel** [comment]
Ok, it seems simple enough to fix as the arc that needs to be created is proportional to the point with the trouble's angle of over 90 (or less than 90 depending on how you look at it).

**Comment by tbfleming** [comment]
Request: make the arc using line segments. My simulator can't handle real arcs.

**Comment by andrewhodel** [comment]
Well, "real arc" just means that the machine converts it to line segments.  For that reason alone I'll only ever make "arcs" as line segments so I can ensure the fragment count.

Circles and polygons with arcs do that in Millcrum.

**Comment by andrewhodel** [comment]
It seems to be just, if the offset path is at an angle of more than N then round the corner based on the bit size.

Or if the bit goes past a point by it's diameter then turn toward the other side (past the angle) of the path and go until that path is intersected.  I'll add that in.
